lib/core/include/scipp/core/element/histogram.h

```diff
@@ -61,9 +62,17 @@ static constexpr auto histogram = overloaded{
         const auto [offset, nbin, scale] = core::linear_edge_params(edges);
         for (scipp::index i = 0; i < scipp::size(events); ++i) {
           const auto x = events[i];
-          const double bin = (x - offset) * scale;
-          if (bin >= 0.0 && bin < nbin)
-            iadd(data, static_cast<scipp::index>(bin), weights, i);
+          scipp::index bin = (x - offset) * scale;
```

Comment:
Is this now performing the `static_cast` implicitly?

Reply:
Yes.